### DISCUSSION TITLE
[讨论] 主页托管在何处最为合适？

### DISCUSSION THREAD

**Discussion by MoYuan-CN** [Meloong-Git/PCL#5014] [自定义主页]
## 🚀 最新信息 🚀
社区投票结束，目前的迁移信息请见
- https://github.com/HomePlaza-Of-PCL2/Discussions/discussions/2

---

### 事件背景

- CloudFlare 在中国大陆地区的连通性一直不太好
近期有报告称部分运营商网络对 CloudFlare 产生 SNI 阻断，导致有的人打不开了……
https://github.com/HomePlaza-Of-PCL2/Homepage-Market/issues/8 
- GitHub 不开加速器或者魔法一直都连不上……
- Gitee 开始对 RAW 的调用产生限制，导致部分主页 403 了（ #5012 ）……
- GitCode 虽说没有限制但很难预计未来是否会有限制（ https://github.com/Hex-Dragon/PCL2/issues/5012#issuecomment-2439399537 ）

### 现在的需求

- 国内稳定的服务商
- 无加速器/魔法情况下的良好连接速度
- 简单，方便的服务管理
- 性价比高的流量费用

### 加入我们的讨论

在下方回复来提供建议，或是加入社区群聊：
- PCL 非官方**主页**社区 @Homeplaza-Of-PCL2 群 719457352
- PCL 非官方社区 @PCL-Community 群 379026872

**Comment by Joker2184** [comment]
之前和凌云讨论过 主页合资化托管.jpg

也来问问

如果你没有足够的资金 但是编写了一个希望给大家帮助的主页 免费托管让你主页访问性一直不好

你会考虑 合资购买 对象存储 搭配 各自主页独享的存储桶 来让国内访问性更好吗

**Reply by wuliaodexiaoluo** [reply]
>你会考虑 合资购买 对象存储 搭配 各自主页独享的存储桶 来让国内访问性更好吗

成本 ++

**Reply by Joker2184** [reply]
目前最便宜且有保障的是雨云

雨云有子面板 或者 干脆直接丢Github 然后自动化部署到存储桶也可以

**Comment by WorldHim** [comment]
家里云（

**Reply by Deep-Dark-Forest** [reply]
V me 50 to buy electricity and Intranet penetration（

**Reply by Joker2184** [reply]
大部分人没有V4公网（

V6的话 有和没有没区别

**Reply by tangge233** [reply]
Try Cloudflare Tunnel
Remember setting your config to IPv6 and HTTP2

**Comment by tangge233** [comment]
如果主页结构不太复杂的话，可以考虑 Cloudflare Worker + 自定义域名

**Reply by Joker2184** [reply]
额（CF访问性不好.jpg...

**Reply by MoYuan-CN** [reply]
见 事件背景 部分第一条，暂不考虑 CloudFlare

**Comment by zkitefly** [comment]
可以试试 gitee pages？

**Reply by Deep-Dark-Forest** [reply]
> 可以试试 gitee pages？

要实名认证……这让不满 16 周岁的开发者变得很困难
并且 gitee pages 好像也不太稳定，参见 PCL2 的帮助文档库的网页版

**Comment by Joker2184** [comment]
emm

拉到100赞助

或者只有付电费和网费的机子（？

看看怎么选

**Comment by zkitefly** [comment]
gitlab pages ?

**Reply by Deep-Dark-Forest** [reply]
那个国内访问也很慢吧……也是境外网站……？

**Comment by zkitefly** [comment]
热铁盒网页托管？

https://host.retiehe.com/

**Reply by Joker2184** [reply]
> 热铁盒网页托管？
> 
> https://host.retiehe.com/

在配置ing 具体信息你可以问问凌云（

**Reply by MoYuan-CN** [reply]
从一开始我就没考虑这个…… 原因是流量大了会被封…… 
我见过食堂六楼里有人被拉小黑屋了……

**Comment by Joker2184** [comment]
[新]更新主页 已进入预览调试阶段

你可以访问 
```
http://pclhomeplazaoss.lingyunawa.top:27000/d/Homepages/UpdateHomepage/UpdateHomepage.xaml 
```
获取主页

**Reply by MoYuan-CN** [reply]
按理来说现在不会出现地区性阻断了…… 测试一周没问题的话我开 Issue 开始改地址

**Reply by llsgllsg** [reply]
我多地ping了一下，还是有解析失败的
![image](https://github.com/user-attachments/assets/88da6a8a-3957-48db-8b8d-3c5ae6ab03b9)

**Reply by Joker2184** [reply]
不应该.jpg

**Reply by Joker2184** [reply]
![mss medicineking top_多地Ping值测试](https://github.com/user-attachments/assets/3849b18c-ee84-4e19-8fba-7051cba70236)
感觉比较随机（？
像突发性的

> 我多地ping了一下，还是有解析失败的 ![image](https://private-user-images.githubusercontent.com/151439431/383179016-88da6a8a-3957-48db-8b8d-3c5ae6ab03b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4OTEzNzAsIm5iZiI6MTczMDg5MTA3MCwicGF0aCI6Ii8xNTE0Mzk0MzEvMzgzMTc5MDE2LTg4ZGE2YThhLTM5NTctNDhkYi04YjhkLTNjNWFlNmFiMDNiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQxMTA0MzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZTI1NTIwMTE0MWE2NmRkYWU1M2Q3ODQzZTE5ZmQ4MzM1Y2Y4NDAwZTY3MWRmODk1YWJkODEwY2YxMWE1MTcwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Dzlj-Htk4wpA52EVOqfEEM1oO8t5YeMoFdC9u_w--PE)

**Comment by wuliaodexiaoluo** [comment]
我有一个大胆的想法，把 xaml 上传到 maven 然后使用阿里云镜像分发，或者传到 npmjs 用 npmmirror.com(
